app/models/concerns/hyrax/collection_behavior.rb

```diff
@@ -16,13 +16,24 @@ module CollectionBehavior
       self.indexer = Hyrax::CollectionIndexer
 
       property :collection_type_gid, predicate: ::RDF::Vocab::SCHEMA.additionalType, multiple: false do |index|
-        index.as :symbol, :stored_searchable
+        index.as :symbol
       end
+
+      after_find { |col| load_collection_type_instance(col) }
+      # before_update { |col| validate_collection_type_gid(col) } # TODO: check that gid is not nil AND check that gid has not changed since it was read in (It is ok to go from nil to a value.)
     end
 
-    attr_accessor :collection_type
     delegate :nestable?, :discoverable?, :sharable?, :allow_multiple_membership?, :require_membership?, :assigns_workflow?, :assigns_visibility?, to: :collection_type
 
+    # Get (and set) the collection_type when accessed
+    def collection_type
+      return @collection_type if @collection_type
+      return nil if collection_type_gid.nil?
+      collection_type = Hyrax::CollectionType.find_by_gid(collection_type_gid)
```

Comment:
I believe it is preferable to replaces lines 32 to 34 with `@collection_type = Hyrax::CollectionType.find_by_gid!`. (Have `.find_by_gid!` raise the consistent exception)